src/zkp/piprm.rs

```diff
@@ -98,20 +96,20 @@ fn generate_challenge_bytes(
     Ok(challenges.into())
 }
 
-impl Proof for PiPrmProof {
-    type CommonInput = RingPedersen;
-    type ProverSecret = PiPrmSecret;
+impl Proof2 for PiPrmProof {
+    type CommonInput<'a> = &'a RingPedersen;
```

Comment:
Does this PR go "deep enough" that is, we accept a `&RingPedersen` vs the exisiting code which just takes a `RingPedersen`. But this still requires the user to have a `RingPedersen` to make a reference out of. Is that fine? One could imagine a `RingPedersen<'a>` where all the fields are references, like we have done with other `CommonInput` types. But I'm not sure that applies here.